tests/plenary/clock/init_spec.lua

```diff
@@ -0,0 +1,52 @@
+local OrgFiles = require('orgmode.files')
+local OrgFile = require('orgmode.files.file')
+local org = require('orgmode')
+
+describe('Clock', function()
+  ---@return OrgFile
+  local load_file_sync = function(content, filename)
+    content = content or {}
+    filename = filename or vim.fn.tempname() .. '.org'
+    vim.fn.writefile(content, filename)
+    return OrgFile.load(filename):wait()
+  end
+
+  it('should properly close out an existing clock when clocking in a new headline', function()
```

Comment:
I think this test is not needed. We already have a similar one. If you want to test some scenario that is not tested you can add a test to that linked file.

Reply:
@kristijanhusak could you please point me to the existing test that is similar? I am not familiar with all the test files so some guidance would be appreciated. This test fails without the patch proposed here, so it does seem to be testing something the other tests are not covering-- keeping the currently clocked header up-to-date as the buffer changes.

Reply:
Ah ok, it's not the same. I didn't notice that you are editing file before clocking in/out.
Please move this test into https://github.com/nvim-orgmode/orgmode/blob/master/tests/plenary/ui/clock_spec.lua and adapt the setup around it. For example, you shouldn't do `OrgFiles:new` directly.